docs/content/guides/dagster/pythonic-config.mdx

```diff
@@ -0,0 +1,246 @@
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_resource_config endbefore=end_basic_resource_config dedent=4
+from dagster import op
+from dagster._config.structured_config import ConfigurableResource
+
+class MyDatabaseResource(ConfigurableResource):
+    connection_url: str
+
+    def query(self, query: str):
+        return get_engine(self.connection_url).execute(query)
+```
+
+For more information on using resources, see [the Pythonic resources guide](/guides/dagster/pythonic-resources).
+
+### Specifying runtime configuration in code
+
+To execute a job or materialize an asset which specifies config, you'll need to provide values for its parameters. When specifying config from the Python API, we can use the `run_config` argument for <PyObject object="JobDefinition" method="execute_in_process"/> or <PyObject object="materialize"/>. This takes a `RunConfig` object, within which we can supply config on a per-op or per-asset basis. The config is specified as a dictionary, with the keys corresponding to the op/asset names and the values corresponding to the config values.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_execute_with_config endbefore=end_execute_with_config dedent=4
+from dagster import job, materialize, op
+
+from dagster._core.definitions.run_config import RunConfig
+
+@job
+def greeting_job():
+    print_greeting()
+
+job_result = greeting_job.execute_in_process(
+    run_config=RunConfig(ops={"print_greeting": MyOpConfig(person_name="Alice")})
+)
+
+asset_result = materialize(
+    [greeting],
+    run_config=RunConfig(ops={"greeting": MyAssetConfig(person_name="Alice")}),
+)
+```
+
+\--
+
+## Defining complex config schemas
+
+In some cases, you may want to define a more complex config schema. For example, you may want to define a config schema that takes in a list of files or complex data. Below we'll walk through some common patterns for defining more complex config schemas.
+
+### Attaching metadata to config fields
+
+Config fields can be annotated with metadata, which can be used to provide additional information about the field, using the Pydantic <PyObject object="Field"/> class.
+
+For example, we can annotate a config field with a description, which will be displayed in the documentation for the config field. We can add a value range to a field, which will be validated when config is specified.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_metadata_config endbefore=end_metadata_config dedent=4
+from dagster._config.structured_config import Config
+from pydantic import Field
+
+class MyMetadataConfig(Config):
+    person_name: str = Field(..., description="The name of the person to greet")
```

Comment:
Can you note in a comma why the `...` is necessary and what the semantic are. I don't really understand it tbh myself.